src/System.Management.Automation/engine/remoting/client/remoterunspaceinfo.cs

```diff
@@ -312,6 +312,17 @@ internal PSSession(RemoteRunspace remoteRunspace)
                 return;
             }
 
+            if (ExperimentalFeature.IsEnabled("PSWinCompat"))
```

Comment:
It would be nice to use the new C# syntax here:
```
switch (remoteRunspace.ConnectionInfo)
{
    case WSManConnectionInfo wsmanConnectionInfo:
        ...
        break;
    
    case NewProcessConnectionInfo newProcessConnectionInfo:
        ....
        break;

    ...
}
```

Reply:
Probably; but that is more like refactoring existing code, which I'd say is out of scope of this PR.